### PR TITLE
DOP-4643: strict rules for image borders

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4643

### Current Behavior:

[previous staged link showed borders around hero image in dark mode](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4643/index.html)

### Staging Links:

[docs landing](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/seung.park/DOP-4643-followup/index.html)
[language pages from docs landing](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/landing/seung.park/DOP-4643-followup/languages/cpp/)
[atlas docs](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4643-followup/index.html)
[atlas docs with image example](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4643-followup/atlas-search/create-index/index.html#node-status)
[regression check](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4643-followup/data-federation/index.html)

### Notes:
- Adds checks to see if this is a landing page or sectionDepth must be greater than 2 to apply border styling in dark mode only

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/components/Image.js [hunk 1]
```diff
@@ -48,32 +65,40 @@ function getImageProps({
     imageProps['height'] = height;
   }
 
+  const applyBorder = hasBorder || (darkMode && getPageSlug(slug) !== '/' && sectionDepth > 1);
```
Comment (mmeigs) [inline src/components/Image.js:68]: Should every dark mode image that passes the second condition get a border? Or do we also need `hasBorder` to be true for the dark mode image to have a border?

Comment (seungpark) [reply src/components/Image.js:68]: this was discussed and updated as the description of the [ticket]( https://jira.mongodb.org/browse/DOP-4643) with discussion to Design team, but also [creating noise for landing page images](https://mongodb.slack.com/archives/C076E2BR90V/p1718205733445249) so i opted to go for non-landing and a farther sectionDepth to be on the cautious side. we did not have any consistent image directive/option for hero image AFAIK

Comment (mmeigs) [reply src/components/Image.js:68]: Sorry, those links aren't really helping me understand that every image in dark mode that isn't a hero should have a border!

Comment (seungpark) [reply src/components/Image.js:66]: thanks for the clarification! removed border (left bordert-radius for rounded images in dark mode)

**Review by mmeigs — APPROVED** [review]
LGTM!

**Pull request merged**